src/NodeServices/PythonServices.cs

```diff
@@ -63,6 +63,13 @@ public abstract object OutputDataMarshaler
             get;
         }
 
+        /// <summary>
+        /// Data marshaler for host data used during the execution of a Python node.
+        /// This can be implemented differently across various Python engines due to
+        /// factors such as differing PythonNet APIs or other specific requirements.
+        /// </summary>
+        public object HostDataMarshaler { get; set; }
```

Comment:
was this moved here from somewhere ? not sure how this is supposed to be used

Reply:
No, this is new. So short answer is : in the same way input and output data marshaler are used.
But I do agree that's not very straight forward either and perhaps at some point some samples should be created at some point. Regardless I'll make this internal since it's more specific to our host products.